File: android/core/clients/src/main/kotlin/com/nexussoft/verbum/clients/BibleComparisonClient.kt
package com.nexussoft.verbum.clients

import com.nexussoft.verbum.clients.helloao.HelloAOBibleClient
import com.nexussoft.verbum.models.*

fun interface BibleComparisonClient { suspend fun passage(reference: PassageReference): BiblePassage }
class LiveBibleComparisonClient : BibleComparisonClient {
    override suspend fun passage(reference: PassageReference): BiblePassage =
        if (BookLanguage.current==BookLanguage.PORTUGUESE) HelloAOBibleClient("por_bsl").passage(reference)
        else BundledBibleClient.passage(reference)
}

File: VerbumKit/Sources/Clients/BibleComparisonClient.swift
import ComposableArchitecture
import Models

@DependencyClient
public struct BibleComparisonClient: Sendable {
    public var passage: @Sendable (_ reference: PassageReference) async throws -> BiblePassage
}
extension BibleComparisonClient: DependencyKey {
    public static let liveValue = Self(passage: { reference in
        if BookLanguage.current == .portuguese {
            return try await HelloAOBibleClient(translationId: "por_bsl").passage(reference)
        }
        return try await BibleClient.bundled.passage(reference: reference)
    })
    public static let previewValue = Self(passage: { reference in try await BibleClient.bundled.passage(reference: reference) })
}
extension DependencyValues {
    public var bibleComparison: BibleComparisonClient {
        get { self[BibleComparisonClient.self] }
        set { self[BibleComparisonClient.self] = newValue }
    }
}
